src/wp-includes/block-template-utils.php

```diff
@@ -513,6 +513,44 @@ function _inject_theme_attribute_in_block_template_content( $template_content )
 	return $template_content;
 }
 
+/**
+ * Returns a function that...
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param WP_Block_Template $block_template a block template.
+ * @return callable A function that returns a block.
+ */
+function _parsed_block_visitor( $block_template ) {
+	return function( $block ) use ( $block_template ) {
+		if (
+			'core/template-part' === $block['blockName'] &&
+			! isset( $block['attrs']['theme'] )
+		) {
+			$block['attrs']['theme'] = get_stylesheet();
+		}
+
+		$hooked_blocks = get_hooked_blocks( $block['blockName'] );
+		foreach ( $hooked_blocks as $hooked_block_type => $relative_position ) {
+			$hooked_block = array(
+				'blockName'    => $hooked_block_type,
+				'attrs'        => array(),
+				'innerHTML'    => '',
+				'innerContent' => array(),
+				'innerBlocks'  => array(),
+			);
+			// Need to pass full current block, (potentially its parent -- for sibiling insertion), relative position, and hooked_block.
+			$block = insert_hooked_block( $hooked_block, $relative_position, $block );
+			/**
+			 * This filter allows modifiying the auto-inserting behavior...
+			 */
+			$block = apply_filters( 'insert_hooked_block', $block, $hooked_blocks[ $hooked_block_type ], $block_template );
```

Comment:
Might use separate filters for template and pattern insertion

```suggestion
			$block = apply_filters( 'insert_hooked_block_into_template', $block, $hooked_blocks[ $hooked_block_type ], $block_template );
```